### ISSUE TITLE
Stochastic Gradient Descent Optimization

### ISSUE THREAD

**Issue by dvigneshwer** [dvigneshwer/deeprust#4]
The cost of running the backpropogation algorithm repeatedly over the whole training
set is quite high, in that case Stochastic Gradient Descent (SGD) has proved to be
a good optimization technique as it can overcome this cost and still lead to fast
convergence as it follows a negative gradient of the objective after seeing only a single
or a few training examples.

The standard gradient descent algorithm updates the parameters θ of the objective
J(θ) as,

**θ = θ − α∆θE[J(θ)]**

where the expectation in the above equation is approximated by evaluating the cost
and gradient over the full training set. SGD reduces the cost by computing the
gradient of the parameters using a small batch of training examples. The new update
is given by,

**θ = θ − α∆θJ(θ; x(i), y(i))**

with a pair (x(i), y(i)) from the training set.

**Comment by dvigneshwer** [comment]
Blog Link : http://dvigneshwer.github.io/blog/2016/12/04/SGD_Optimizer
